akkeeper/src/main/scala/akkeeper/config/RichConfig.scala

```diff
@@ -15,102 +15,133 @@
+    if (launcherConfig.hasPath("timeout")) {
+      Some(launcherConfig.getDuration("timeout", TimeUnit.SECONDS).seconds)
     } else {
-      Seq.empty
+      None
     }
   }
 
-  def getActorSystemName: String = {
-    config.getString("akkeeper.akka.system-name")
-  }
+  // Zookeeper configs
+  def zookeeperClientConfig: ZookeeperClientConfig = ZookeeperClientConfig.fromConfig(zookeeperConfig)
 
-  def getYarnConfig: Config = {
-    config.getConfig("akkeeper.yarn")
-  }
+  // Rest API configs
+  def restPort: Int = restConfig.getInt("port")
+  def restPortMaxAttempts: Int = restConfig.getInt("port-max-attempts")
+  def restRequestTimeout: Timeout = Timeout(
+    restConfig.getDuration("request-timeout", TimeUnit.MILLISECONDS),
+    TimeUnit.MILLISECONDS)
 
-  def getYarnApplicationName: String = {
-    getYarnConfig.getString("application-name")
-  }
+  // Kerberos configs
+  def kerberosTicketCheckInterval: Long =
+    kerberosConfig.getDuration("ticket-check-interval", TimeUnit.MILLISECONDS)
 
-  def getYarnStagingDirectory(conf: Configuration, appId: String): String ={
+  // Yarn configs
+  def yarnApplicationName: String = yarnConfig.getString("application-name")
+  def yarnJvmArgs: mutable.Seq[String] = yarnConfig.getStringList("master.jvm.args").asScala
+  def yarnMaxAttempts: Int = yarnConfig.getInt("max-attempts")
+  def yarnClientThreads: Int = yarnConfig.getInt("client-threads")
+  def yarnStagingDirectory(conf: Configuration, appId: String): String ={
     val basePath =
-      if (getYarnConfig.hasPath("staging-directory")) {
-        getYarnConfig.getString("staging-directory")
+      if (yarnConfig.hasPath("staging-directory")) {
+        yarnConfig.getString("staging-directory")
       } else {
         new Path(FileSystem.get(conf).getHomeDirectory, ".akkeeper").toString
       }
     new Path(basePath, appId).toString
   }
 
-  def getZookeeperClientConfig: ZookeeperClientConfig = {
-    ZookeeperClientConfig.fromConfig(config.getConfig("akkeeper.zookeeper"))
+  // Yarn master configs
+  def yarnMasterCpus: Int = yarnMaster.getInt("cpus")
+  def yarnMasterMemory: Int = yarnMaster.getInt("memory")
+
+  // Other
+  def withMasterRole: Config = {
+    config.withValue("akka.cluster.roles",
+      ConfigValueFactory.fromIterable(Seq(MasterService.MasterServiceName).asJava))
   }
 
-  def getContainers: Seq[ContainerDefinition] = {
-    if (config.hasPath("akkeeper.containers")) {
-      val configContainers = config.getConfigList("akkeeper.containers").asScala
-      configContainers.map(ContainerDefinition.fromConfig)
+  def withMasterPort: Config = {
+    config.withValue("akka.remote.netty.tcp.port", ConfigValueFactory.fromAnyRef(akkaPort))
+  }
+
+  def withPrincipalAndKeytab(principal: String, keytab: String): Config = {
+    config
+      .withValue("akka.kerberos.principal", ConfigValueFactory.fromAnyRef(principal))
+      .withValue("akka.kerberos.keytab", ConfigValueFactory.fromAnyRef(keytab))
+  }
+
+  // Utils
```

Comment:
I suggest to make all "utils" method private.

Reply:
They are used from outside :(